class.jetpack-autoupdate.php

```diff
@@ -48,5 +55,60 @@ function autoupdate_core( $update, $item ) {
 		}
 		return $update;
 	}
+
+	/**
+	 * Filter for wp_get_update_data that doesn't actually filter anything
+	 * Used to save a Jetpack_Option when ever new updates are detected
+	 *
+	 */
+	function save_update_data( $update_data ) {
+		global $wp_version;
+
+		// If we are not on the main site, no need to save.
+		if ( ! is_main_site() ) {
+			return $update_data;
+		}
+
+		if ( isset( $update_data['counts'] ) ) {
+			$updates = $update_data['counts'];
+		}
+
+		$updates['wp_version'] = isset( $wp_version ) ? $wp_version : null;
+
+		if ( ! empty( $updates['wordpress'] ) ) {
+			$cur = get_preferred_from_update_core();
+			if ( isset( $cur->response ) && $cur->response === 'upgrade' ) {
+				$updates['wp_update_version'] = $cur->current;
+			}
+		}
+
+		$updates['site_is_vcs'] = (bool) $this->is_vcs();
+		Jetpack_Options::update_option( 'updates', $updates );
+		return $update_data;
+	}
+
+	/**
+	 * Finds out if a site is using a version control system.
+	 * We'll store that information as a transient with a 24 expiration.
+	 * We only need to check once per day.
+	 *
+	 * @return string ( '1' | '0' )
+	 */
+	function is_vcs() {
+		$is_vcs = get_transient( 'jetpack_site_is_vcs' );
+
+		if ( false === $is_vcs ) {
+			include_once ABSPATH . 'wp-admin/includes/class-wp-upgrader.php';
```

Comment:
We should check if we even need to include it. 
So if ( ! class_exists( 'WP_Automatic_Updater' ) )  {
include_once ABSPATH . 'wp-admin/includes/class-wp-upgrader.php';
}